### PR TITLE
feat(cli): Manage Alert Rules in Lacework CLI

### PR DESCRIPTION
**Pull request by dmurray-lacework**
## Summary

Manage Alert Rules in the Lacework cli

## How did you test this change?
New integration tests `integration/alert_rules_test.go`
Manually run `lacework alert-rule create | list | show | delete`

## Issue

https://lacework.atlassian.net/browse/ALLY-681

### CODE REVIEW

File: cli/cmd/alert_rules.go [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/lacework/go-sdk/api"
+	"github.com/olekukonko/tablewriter"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
```
Comment (afiune) [inline cli/cmd/alert_rules.go:31]: nit: if you run `make fmt` which runs `goimports` it should group imports like this:
```suggestion
	"github.com/AlecAivazis/survey/v2"
	"github.com/olekukonko/tablewriter"
	"github.com/pkg/errors"
	"github.com/spf13/cobra"
	
	"github.com/lacework/go-sdk/api"
```

File: cli/cmd/alert_rules.go [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+			if !cli.InteractiveMode() {
+				return errors.New("interactive mode is disabled")
+			}
+
+			response, err := promptCreateAlertRule()
+			if err != nil {
+				return errors.Wrap(err, "unable to create alert rule")
+			}
+
+			cli.OutputHuman(fmt.Sprintf("The alert rule was created with ID %s \n", response.Data.Guid))
```
Comment (afiune) [inline cli/cmd/alert_rules.go:143]: in the table we call this column `GUID`, so update this or the table.
```suggestion
			cli.OutputHuman(fmt.Sprintf("The alert rule was created with GUID '%s'.\n", response.Data.Guid))
```

File: cli/cmd/alert_rules.go [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+			if cli.JSONOutput() {
+				return cli.OutputJSON(alertRules)
+			}
+
+			var rows [][]string
+			for _, rule := range alertRules.Data {
+				rows = append(rows, []string{rule.Guid, rule.Filter.Name, rule.Filter.Description, rule.Filter.Status()})
+			}
+
+			cli.OutputHuman(renderSimpleTable([]string{"GUID", "NAME", "DESCRIPTION", "ENABLED"}, rows))
```
Comment (afiune) [inline cli/cmd/alert_rules.go:80]: I think the description could be too long for displaying it in the `list` command. Should we NOT display it and let the user run a `show` command for showing more details?
```suggestion
			cli.OutputHuman(renderSimpleTable([]string{"GUID", "NAME", "ENABLED"}, rows))
```

Comment (afiune) [reply cli/cmd/alert_rules.go:80]: Maybe instead, we could show the severities? 🤔  Thoughts?

File: cli/cmd/alert_rules.go [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+			}),
+		),
+		tableFunc(func(t *tablewriter.Table) {
+			t.SetBorder(false)
+			t.SetAutoWrapText(false)
+		}),
+	),
+	)
+
+	detailsTable.WriteString(renderCustomTable([]string{"CHANNELS", "RESOURCE GROUPS"}, IDsList,
```
[the remark this answers is not in the document]
Comment (afiune) [reply cli/cmd/alert_rules.go:194]:
```suggestion
	detailsTable.WriteString(renderCustomTable([]string{"ALERT CHANNELS", "RESOURCE GROUPS"}, IDsList,
```

File: cli/cmd/alert_rules.go [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+		channels = append(channels, displayName)
+		items[displayName] = i.ID()
+	}
+
+	return channels, items
+}
+
+func getResourceGroups() ([]string, map[string]string) {
+	response, err := cli.LwApi.V2.ResourceGroups.List()
+	var groups []string
```
Comment (afiune) [inline cli/cmd/alert_rules.go:310]: nit: same here, the groups variable is not used if there is an error.

**Review by afiune — CHANGES_REQUESTED** [review]
Looking good!

tenor-187130103

File: cli/cmd/alert_rules.go (hunk 3 above)
Comment (dmurray-lacework) [reply cli/cmd/alert_rules.go:80]: I don't think severities would fit well with the "headers". IMO it should live in "details"

File: cli/cmd/alert_rules.go [hunk 6]
```diff
@@ -0,0 +1,372 @@
...
+		Use:     "alert-rule",
+		Aliases: []string{"alert-rules", "ar"},
+		Short:   "manage alert rules",
+		Long: `Manage routing events to channels using alert rules.
+Configure alert rules with alert channels and optionally resource groups
+to send events in one of these categories: 
+App, Compliance, Cloud, File, Machine, User and Platform
+with any of these severities:
+Critical, High, Medium, Low and Info
+`,
```
Comment (afiune) [inline cli/cmd/alert_rules.go:47]:
```suggestion
		Long: `Manage alert rules to route events to the appropriate people or tools.		

An alert rule has three parts:
  1. Alert channel(s) that should receive the event notification
  2. Event severity and categories to include
  3. Resource group(s) containing the subset of your environment to consider
`,
```

File: cli/cmd/alert_rules.go [hunk 7]
```diff
@@ -0,0 +1,350 @@
...
+		Use:   "delete",
+		Short: "delete a alert rule",
+		Long:  "Delete a single alert rule by it's ID.",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			err := cli.LwApi.V2.AlertRules.Delete(args[0])
+			if err != nil {
+				return errors.Wrap(err, "unable to delete alert rule")
+			}
+			return nil
```
[the remark this answers is not in the document]
Comment (afiune) [reply cli/cmd/alert_rules.go:131]: @dmurray-lacework can we still print a message like `The alert rule ID was deleted`?

File: cli/cmd/alert_rules.go [hunk 8]
```diff
@@ -0,0 +1,372 @@
...
+		tableFunc(func(t *tablewriter.Table) {
+			t.SetBorder(false)
+			t.SetAutoWrapText(false)
+		}),
+	),
+	)
+
+	if len(rule.Channels) > 0 {
+		channels := [][]string{{strings.Join(rule.Channels, "\n")}}
+		detailsTable.WriteString(renderCustomTable([]string{"CHANNELS"}, channels,
```
Comment (afiune) [inline cli/cmd/alert_rules.go:202]:
```suggestion
		detailsTable.WriteString(renderCustomTable([]string{"ALERT CHANNELS"}, channels,
```

File: cli/cmd/alert_rules.go [hunk 9]
```diff
@@ -0,0 +1,372 @@
...
+	}
+
+	return detailsTable.String()
+}
+
+func promptCreateAlertRule() (api.AlertRuleResponse, error) {
+	channelList, channelMap := getAlertChannels()
+
+	if len(channelList) < 1 {
+		return api.AlertRuleResponse{}, errors.New("No Alert Channels found.")
```
Comment (afiune) [inline cli/cmd/alert_rules.go:230]: I am not sure I like the capitalization in this error message:
```suggestion
		return api.AlertRuleResponse{}, errors.New("no alert channels found")
```

**Review by afiune — CHANGES_REQUESTED** [review]

**Review by afiune — APPROVED** [review]
tenor-18739220

**Pull request merged**